interface/src/app/mlmodel/mlModelComponents.js

```diff
@@ -30,4 +31,45 @@ angular.module('adage.mlmodel.components', [
   }]
 })
 
+.component('mlModelView', {
+  templateUrl: 'mlmodel/view.tpl.html',
+  controller: ['GlobalModelInfo', function(GlobalModelInfo) {
+    this.modelInfo = GlobalModelInfo;
+  }]
+})
+
+.component('mlModelValidator', {
```

Comment:
This component not only validates the input ML model ID, it is also responsible for generating an error message on the web UI. Please read the long comments that I wrote 6 days ago and see whether it makes sense to you.